lib/octopus_auth/access_scope_validator.rb

```diff
@@ -0,0 +1,24 @@
+module OctopusAuth
+  class AccessScopeValidator
+    def initialize(access_token)
+      @access_token = access_token
+      @access_scopes = (access_token.access_scopes || '').split(OctopusAuth.configuration.access_scopes_delimiter)
+    end
+
+    def valid?(*required_scopes)
+      access_all_scopes? || required_scopes.all? { |scope| access_scopes.include?(scope.to_s) }
+    end
+
+    def self.valid?(access_token, *required_scopes)
+      self.new(access_token).valid?(*required_scopes)
+    end
+
+    private
+
+    attr_reader :access_token, :access_scopes
+
+    def access_all_scopes?
+      access_scopes.any? { |scope| scope.to_s == OctopusAuth.configuration.access_scopes_wildcard.to_s }
```

Comment:
I think `scope.to_s` is redundant. `scope` is result of `split`, which should be string already.

Reply:
You could post-process `configuration.access_scopes_wildcard` in `/lib/octopus_auth.rb:21` to make sure it's always a string to avoid `to_s`